src/main/groovy/io/gitlab/grabl/CreateDatabase.groovy

```diff
@@ -0,0 +1,125 @@
+package io.gitlab.grabl
+
+import org.gradle.api.tasks.Input
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.Optional
+
+class CreateDatabase extends DefaultTask {
+
+    // If there's an easier way to do this WHILE keeping the hard-typing
+    // functionality, let aestrada@progress.com know
+    @Input
+    String dbName
+
+    @Input @Optional
+    String destDir = null
+
+    @Input @Optional
+    String sourceDb = null
+
+    @Input @Optional
+    String schemaFile = null
+
+    @Input @Optional
+    String structFile = null
+
+    @Input @Optional
+    int blockSize = 0
+
+    @Input @Optional
+    boolean noInit = null
+
+    @Input @Optional
+    String codepage = null
+
+    @Input @Optional
+    String wordRules = null
+
+    @Input @Optional
+    boolean multiTenant = null
+
+    @Input @Optional
+    boolean failOnError = null
+
+    @Input @Optional
+    String collation = null
+
+    @Input @Optional
+    String tempDir = null
```

Comment:
Probably should be @InputFile

Reply:
or @InputDir

Reply:
It doesn't matter, since this is being passed to ant, which expects a String, not a file.

Reply:
It's extra steps since ant.PCTCreateDatabase does not want gradle File object. If you do:

    def f = new File("something")

    ant.PCTCreateBase(dbName = "hello.db", tempDir = f)

It won't work, you have to do:

    ant.PCTCreateBase(dbName = "hello.db", tempDir = "${f}")

which converts the file to its String form.

Reply:
So if we do change the type to File, people would have to stop doing this:

    task foo(type: io.gitlab.grabl.CreateDatabase) {
        tempDir = "/tmp"
    }

And do this instead:

    task foo(type: io.gitlab.grabl.CreateDatabase) {
        tempDir = new File("/tmp")
    }

Which is stricter, but we don't really gain anything from doing this. We don't check to see if this folder exists, we don't check permissions, etc. The only thing we possibly gain is we get an error if a string is passed saying "tempDir was passed a String when it expected an File". Which is kinda useless since all we do is convert it to a String again anyway.

So sure, we can make it a File, I just didn't really see any real reason to do it but if you think we really should, then it's an easy conversion.